File: omnipod-dash/src/main/java/info/nightscout/androidaps/plugins/pump/omnipod/dash/driver/comm/CharacteristicType.java
package info.nightscout.androidaps.plugins.pump.omnipod.dash.driver.comm;

import java.math.BigInteger;
import java.util.UUID;

public enum CharacteristicType {
    CMD("1a7e-2441-e3ed-4464-8b7e-751e03d0dc5f"),
    DATA("1a7e-2442-e3ed-4464-8b7e-751e03d0dc5f");

    public final String value;

    CharacteristicType(String value) {
        this.value = value;
    }

    public static CharacteristicType byValue(byte value) {
        for (CharacteristicType type : values()) {
            if (type.value.equals(value)) {
                return type;
            }
        }
        throw new IllegalArgumentException("Unknown Characteristic Type: " + value);
    }

    public String getValue() {
        return this.value;
    }

    public UUID getUUID() {
        return new UUID(
                new BigInteger(this.value.replace("-", "").substring(0, 16), 16).longValue(),
                new BigInteger(this.value.replace("-", "").substring(16), 16).longValue()
        );
    }
}

File: pump/omnipod/dash/src/main/kotlin/app/aaps/pump/omnipod/dash/driver/comm/io/CharacteristicType.kt
package app.aaps.pump.omnipod.dash.driver.comm.io

import java.math.BigInteger
import java.util.*

enum class CharacteristicType(val value: String) {
    CMD("1a7e2441-e3ed-4464-8b7e-751e03d0dc5f"), DATA("1a7e2442-e3ed-4464-8b7e-751e03d0dc5f");

    val uuid: UUID
        get() = UUID(
            BigInteger(value.replace("-", "").substring(0, 16), 16).toLong(),
            BigInteger(value.replace("-", "").substring(16), 16).toLong()
        )

    companion object {

        fun byValue(value: String): CharacteristicType =
            CharacteristicType.entries.firstOrNull { it.value == value }
                ?: throw IllegalArgumentException("Unknown Characteristic Type: $value")
    }
}
